### ISSUE TITLE
Possible bug

### ISSUE THREAD

**Issue by Rewbert** [augustss/MicroHs#63]
Consider this program
```haskell
module Example(main) where

main :: IO ()
main = do
  let x :: Int
      (_, x) = (10, 5)
  print x
```
It compiles perfectly well with GHC (with or without the Haskell98 flag), but not with MicroHaskell. If I remove the type signature, it compiles with MicroHaskell.

**Comment by augustss** [comment]
Yeah, that's a bug.

**Comment by augustss** [comment]
Fixed in 2cb9870808f0c9b8996ffc6c148d722d3b89b34c
